### PR TITLE
fix workload status for multiple revision case

### PR DESCRIPTION
**Pull request by wonderflow**
For revisionEnabled case, oam runtime will generate multiple workload, while only one workload per component will list in status of AppConfig.

This PR will fix it and list all workloads in status of appconfig.

Signed-off-by: 天元 <jianbo.sjb@alibaba-inc.com>

### CODE REVIEW

File: test/e2e-test/suite_test.go [hunk 1]
```diff
@@ -218,6 +218,51 @@ var _ = BeforeSuite(func(done Done) {
 	}
 	Expect(k8sClient.Create(context.Background(), &crd)).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
 	By("Created a crd for appconfig dependency test")
+
+	crd = crdv1.CustomResourceDefinition{
```
Comment (ryanzhang-oss) [inline test/e2e-test/suite_test.go:222]: Why can't this test re-use the crd for the appconfig dependency test?

File: test/e2e-test/component_version_test.go [hunk 2]
```diff
@@ -297,6 +303,182 @@ var _ = Describe("Versioning mechanism of components", func() {
...
+				time.Second*15, time.Millisecond*500).Should(BeEquivalentTo("v2"))
+
+			By("Check AppConfig status")
+			Eventually(
+				func() error {
+					return k8sClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: appconfig.Name}, &appconfig)
+				},
+				time.Second*15, time.Millisecond*500).Should(BeNil())
+
+			Expect(len(appconfig.Status.Workloads)).Should(BeEquivalentTo(1))
```
Comment (ryanzhang-oss) [inline test/e2e-test/component_version_test.go:468]: Why not check that its revision name should be v2 too

File: test/e2e-test/component_version_test.go [hunk 3]
```diff
@@ -297,6 +303,182 @@ var _ = Describe("Versioning mechanism of components", func() {
...
+
+			//Clean
+			k8sClient.Delete(ctx, &appconfig)
+			k8sClient.Delete(ctx, &comp1)
+			k8sClient.Delete(ctx, &comp2)
+		})
+	})
 })
+
+func readYaml(path string, object runtime.Object) error {
```
[the remark this answers is not in the document]
Comment (wonderflow) [reply test/e2e-test/component_version_test.go:478]: it's a small function, and I didn't see we need to use it in our business code, so I leave it here

File: test/e2e-test/suite_test.go (hunk 1 above)
Comment (wonderflow) [reply test/e2e-test/suite_test.go:222]: I'm afraid the Definition（cluster scope CRD） file will conflict, so I create a new one.

File: test/e2e-test/component_version_test.go (hunk 2 above)
Comment (wonderflow) [reply test/e2e-test/component_version_test.go:468]: sure

Comment (wonderflow) [reply test/e2e-test/component_version_test.go:468]: I have checked it in line 459

File: pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go [hunk 4]
```diff
@@ -313,6 +312,43 @@ func (r *OAMApplicationReconciler) Reconcile(req reconcile.Request) (result reco
...
+		if err := r.client.List(ctx, &ul, client.MatchingLabels{oam.LabelAppName: ac.Name, oam.LabelAppComponent: w.ComponentName}); err != nil {
+			continue
+		}
+		for _, v := range ul.Items {
+			if v.GetName() == w.ComponentRevisionName {
+				continue
+			}
+			// These workload exists means the component is under progress of rollout
+			// Trait will not work for these remaining workload
+			revisionStatus = append(revisionStatus, v1alpha2.WorkloadStatus{
```
[the remark this answers is not in the document]
Comment (wonderflow) [reply pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go:335]: Sure, I can add this flag

File: test/e2e-test/component_version_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryanzhang-oss) [reply test/e2e-test/component_version_test.go:478]: at least make this available for other tests?

File: test/e2e-test/suite_test.go (hunk 1 above)
Comment (ryanzhang-oss) [reply test/e2e-test/suite_test.go:222]: how do they conflict?

Comment (wonderflow) [reply test/e2e-test/suite_test.go:222]: using it as traitdefinition, revisionEnabled true/false will affect the test result.

File: test/e2e-test/component_version_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (wonderflow) [reply test/e2e-test/component_version_test.go:480]: Let's put it next time if we need it

**Pull request merged**